File: src/main/resources/output/IFC4X3_csharp/Entities/IfcRoot/IfcObjectDefinition/IfcObject/IfcProduct/IfcStructuralItem/IfcStructuralMember/IfcStructuralCurveMember/IfcStructuralCurveMember.cs
public class IfcStructuralCurveMember : IfcStructuralMember
{
    public IfcStructuralCurveMemberTypeEnum PredefinedType { get; set; }
    public IfcDirection Axis { get; set; }

    // === EXTENDED BY ===
    // IfcStructuralCurveMemberVarying

    // === WHERE CLAUSES ===
    // HasObjectType : (PredefinedType <> IfcStructuralCurveMemberTypeEnum.USERDEFINED) OR EXISTS(SELF\IfcObject.ObjectType)
}

File: output/ifc4x3_dev_73740fe4/entity/IfcStructuralCurveMember.java
package com.topoplanet.ifc4x3_dev_73740fe4.entity;
import com.topoplanet.test.type.*;


public class IfcStructuralCurveMember extends IfcStructuralMember {

    public IfcStructuralCurveMemberTypeEnum PredefinedType;
    public IfcDirection Axis;
    // INVERSE attributes:
    // HasObjectType : (PredefinedType <> IfcStructuralCurveMemberTypeEnum.USERDEFINED) OR EXISTS(SELF\IfcObject.ObjectType);

    // SUPERTYPE OF:
    // - IfcStructuralCurveMemberVarying SUBTYPE OF IfcStructuralMember

    // WHERE constraints:
    // HasObjectType : (PredefinedType <> IfcStructuralCurveMemberTypeEnum.USERDEFINED) OR EXISTS(SELF\IfcObject.ObjectType);
}
